lib/bolt/plugin/terraform.rb

```diff
@@ -49,11 +71,32 @@ def inventory_targets(opts)
       end
 
       def load_statefile(opts)
+        statefile = if opts['backend'] == 'remote'
+                      load_remote_statefile(opts)
+                    else
+                      load_local_statefile(opts)
+                    end
+
+        JSON.parse(statefile)
+      end
+
+      # Uses the Terraform CLI to pull remote state files
+      def load_remote_statefile(opts)
+        stdout_str, stderr_str, = Open3.capture3('terraform state pull', chdir: opts['dir'])
```

Comment:
Would it be worth wrapping this in a begin/rescue block and raising a bolt error if terraform executable is not on PATH? Also is there anything useful we would want to do with the exit code?

Reply:
Added a begin/rescue block - good catch. Also modified the error after the block to read the status code instead of `stderr`.